dask/tests/test_spark_compat.py

```diff
@@ -0,0 +1,108 @@
+import signal
+import sys
+import threading
+
+import pytest
+
+from dask.datasets import timeseries
+
+dd = pytest.importorskip("dask.dataframe")
+pyspark = pytest.importorskip("pyspark")
+
+from dask.dataframe.utils import assert_eq
+
+pytestmark = pytest.mark.spark
+
+# pyspark auto-converts timezones -- round-tripping timestamps is easier if
+# we set everything to UTC.
+pdf = timeseries(freq="1H").compute()
+pdf.index = pdf.index.tz_localize("UTC")
+pdf = pdf.reset_index()
+
+
+@pytest.fixture(scope="module")
+def spark_session():
+    if not sys.platform.startswith("linux"):
+        pytest.skip(
+            "Unnecessary, and hard to get spark working on non-linux platforms",
+        )
+
+    # Spark registers a global signal handler that can cause problems elsewhere
+    # in the test suite. In particular, the handler fails if the spark session
+    # is stopped (a bug in pyspark).
+    prev = signal.getsignal(signal.SIGINT)
+    # Create a spark session. Note that we set the timezone to UTC to avoid
+    # conversion to local time when reading parquet files.
+    spark = (
+        pyspark.sql.SparkSession.builder.master("local")
+        .appName("Dask Testing")
+        .config("spark.sql.session.timeZone", "UTC")
+        .getOrCreate()
+    )
+    yield spark
+
+    spark.stop()
```

Comment:
As far as I can tell, the pyspark session interface does not expose a context manager (indeed, I had to do the extra stuff with signal handlers because it generally does a poor job of cleaning up after itself)